### PR TITLE
Migrate to final `halo2_gadgets` pre-release revision

### PR DESCRIPTION
**Pull request by str4d**
Includes API changes made in zcash/halo2#573.

### CODE REVIEW

File: src/circuit.rs [hunk 1]
```diff
@@ -438,18 +436,30 @@ impl plonk::Circuit<pallas::Base> for Circuit {
...
+                v_net,
+                rcv,
             )?;
 
             // Constrain cv_net to equal public input
             layouter.constrain_instance(cv_net.inner().x().cell(), config.primary, CV_NET_X)?;
             layouter.constrain_instance(cv_net.inner().y().cell(), config.primary, CV_NET_Y)?;
 
-            v_net
+            // Return the magnitude and sign so we can use them in the Orchard gate.
```
Comment (ebfull) [inline src/circuit.rs:461]: nonblocking: What's the "Orchard gate"

File: Cargo.toml [hunk 2]
```diff
@@ -83,7 +83,7 @@ debug = true
 debug = true
 
 [patch.crates-io]
-halo2_gadgets = { git = "https://github.com/zcash/halo2.git", rev = "30f92f3f4b785ea2a32392bf65c1b08f0411567c" }
-halo2_proofs = { git = "https://github.com/zcash/halo2.git", rev = "30f92f3f4b785ea2a32392bf65c1b08f0411567c" }
+halo2_gadgets = { git = "https://github.com/zcash/halo2.git", rev = "4b802a7d07b4eefeca137e4f0939ce8e13e0b954" }
+halo2_proofs = { git = "https://github.com/zcash/halo2.git", rev = "4b802a7d07b4eefeca137e4f0939ce8e13e0b954" }
```
Comment (ebfull) [inline Cargo.toml:87]: Checked that these match the upstream PR.

**Review by ebfull — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (ebfull) [reply Cargo.toml:87]: Actually we'll want to change these to the post-merge commit hashes.

File: src/circuit.rs (hunk 1 above)
Comment (str4d) [reply src/circuit.rs:461]: The gate controlled by `q_orchard`, defined earlier.

File: Cargo.toml (hunk 2 above)
Comment (str4d) [reply Cargo.toml]: Done.

**Review by ebfull — APPROVED** [review]

File: src/circuit.rs [hunk 3]
```diff
@@ -492,7 +505,11 @@ impl plonk::Circuit<pallas::Base> for Circuit {
         let pk_d_old = {
             let ivk = {
                 let ak = ak_P.extract_p().inner().clone();
-                let rivk = self.rivk.map(|rivk| rivk.inner());
+                let rivk = ScalarFixed::new(
+                    ecc_chip.clone(),
+                    layouter.namespace(|| "rcv"),
```
Comment (daira) [inline src/circuit.rs:510]: `layouter.namespace(|| "rivk"),`

Comment (str4d) [reply src/circuit.rs:510]: #329

File: src/circuit.rs [hunk 4]
```diff
@@ -586,7 +607,11 @@ impl plonk::Circuit<pallas::Base> for Circuit {
                 self.psi_new,
             )?;
 
-            let rcm_new = self.rcm_new.as_ref().map(|rcm_new| rcm_new.inner());
+            let rcm_new = ScalarFixed::new(
+                ecc_chip,
+                layouter.namespace(|| "rcm_old"),
```
Comment (daira) [inline src/circuit.rs:612]:
```rust
layouter.namespace(|| "rcm_new"),
```

**Review by daira — COMMENTED** [review]
Post-hoc utACK. There are two copy-paste errors in annotations.

**Pull request merged**